tests/Issues/Issue8752/Source/RemoveDoctrineAnnotationValueRector.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\Issues\Issue8752\Source;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocTagNode;
+use Rector\BetterPhpDocParser\PhpDoc\DoctrineAnnotationTagValueNode;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory;
+use Rector\Comments\NodeDocBlock\DocBlockUpdater;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+class RemoveDoctrineAnnotationValueRector extends AbstractRector
+{
+    public function __construct(
+        private readonly PhpDocInfoFactory $phpDocInfoFactory,
+        private readonly DocBlockUpdater $phpDocUpdater,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('// @todo fill the description', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+/**
+ * @OA\Schema(
+ *     schema="ContentBlockTreeResponseDTO",
+ * )
+ */
+class DoctrineAnnotationWithValue
+{
+}
+
+?>
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+/**
+ * @OA\Schema
+ */
+class DoctrineAnnotationWithValue
+{
+}
+CODE_SAMPLE,
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNode($node);
+        if (!$phpDocInfo instanceof PhpDocInfo) {
+            return null;
+        }
+
+        foreach ($phpDocInfo->getPhpDocNode()->children as $phpDocChildNode) {
+            if (!$phpDocChildNode instanceof PhpDocTagNode) {
+                continue;
+            }
+            if (!$phpDocChildNode->value instanceof DoctrineAnnotationTagValueNode) {
+                continue;
+            }
+
+            $doctrineTagValueNode = $phpDocChildNode->value;
+            foreach ($doctrineTagValueNode->getValues() as $value) {
+                $doctrineTagValueNode->removeValue($value->key);
+            }
```

Comment:
For your use case, the solution I found is just replace the object :)

```suggestion
            $phpDocChildNode->value = new DoctrineAnnotationTagValueNode(
                $phpDocChildNode->value->identifierTypeNode
            );
```

Reply:
@samsonasik Thank you for your involvement.

The main problem I'm trying to solve is grouping positional values ​​into named. The algorithm is as follows:

1. Collect all values that match some pattern;
2. Remove them from original node;
3. Add new named parameter with collected values.

I want to transform

```php
/**
 * @OA\Schema(
 *     schema="ContentBlockTreeResponseDTO",
 *     @OA\Property(property="id", type="integer"),
 *     @OA\Property(property="name", type="string")
 * )
 */
```

into

```php
/**
 * @OA\Schema(
 *     schema="ContentBlockTreeResponseDTO",
 *     properties={@OA\Property(property="id", type="integer"), @OA\Property(property="name", type="string")}
 * )
 */
```

In this PR I just wanted to show that a public API of `AbstractValuesAwareNode` is not enough to perform manipulations to delete values ​​in some cases. One of the small problems that I encountered when solving my main problem.

Could you give me some advices how to solve my main problem? Should I build new node?

Reply:
It seems harder than I can think, but more test case would be great so it may slowly can be resolved 👍

Reply:
also, feel free to provide a patch :)

Reply:
@pincher2012 I'm just looking into this and running your test locally. It's well written :+1: 

As your requirement are quite complex, it would be scratching behind ear to make an universal API from it.

After bit of experimenting, I'd take the way @samsonasik suggests. Instead of trimming of existing docblocks, I'd do it in 2 steps: 

* collect needed data, 
* and create a new docblock that fits your needs
 
You can merge them into single named easily :+1: